### ISSUE TITLE
Async shortcodes in includes in for loops lead to empty output

### ISSUE THREAD

**Issue by ryanccn** [11ty/eleventy#2944]
### Operating system

macOS Ventura 13.3.1

### Eleventy

2.0.1

### Describe the bug

When an include calls an async shortcode and that include is then included in a for loop (in Nunjucks, haven't tested in other templating languages), the for loop ends up not having any output.

The shortcode is run but the output doesn't seem to be awaited, and any other content in the for loop also get discarded.

### Reproduction URL

https://github.com/ryanccn/eleventy-includes-async-shortcodes-demo

**Comment by ryanccn** [comment]
Thanks to @uncenter for helping me locating the reason for this bug

**Comment by ryanccn** [comment]
Seems to be fixed by using `asyncEach` or `asyncAll`?

**Comment by ryanccn** [comment]
Maybe it would be better to mention this in the docs about asynchronous shortcodes

**Comment by pdehaan** [comment]
Yeah, it's a bit buried in the Nunjucks docs at https://mozilla.github.io/nunjucks/templating.html#asynceach:
> _"At compile-time, Nunjucks is not aware how templates are loaded so it's unable to determine if an `include` block is asynchronous or not. That's why it can't automatically convert loops for you, and you must use `asyncEach` for iteration if you are loading templates asynchronously inside the loop."_

**Comment by zachleat** [comment]
Moved this one to #4308 as the issue has pivoted. Our upcoming Nunjucks v4 fork (coming in Core v4.0.0-alpha.9) will be fully async-friendly and you won’t have to use special tags to handle async behavior
